### PR TITLE
Fix collections.abc imports for Python 3.9

### LINKED ISSUE #7692: Properly use collections.abc
Python 3.9 is planning on enforcing that abstract base classes in the collections module are imported from `collections.abc`. See https://docs.python.org/3/library/collections.html#module-collections.

We currently are not always doing this in Certbot and are ignoring the deprecation warning about it at https://github.com/certbot/certbot/blob/6e07e8b5c0f3c6da2ae03a2a63adc7ce8e15a202/pytest.ini#L16

Properly testing and finding all occurrences of this problem is a little tricky (see https://github.com/certbot/certbot/issues/7691), but we should fix this and remove the `pytest.ini` entry well before Python 3.9 is released.

### PR DESCRIPTION
**Pull request by hugovk**
## Pull Request Checklist

- [x] If the change being made is to a [distributed component](https://certbot.eff.org/docs/contributing.html#code-components-and-layout), edit the `master` section of `certbot/CHANGELOG.md` to include a description of the change being made.
- [x] Include your name in `AUTHORS.md` if you like.

Fixes #7692.

What sort of `certbot/CHANGELOG.md`

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -10,4 +10,3 @@
 filterwarnings =
     error
     ignore:decodestring:DeprecationWarning
-    ignore:.*collections\.abc:DeprecationWarning
```
Comment (tirkarthi) [inline pytest.ini]: Nit : The text at https://github.com/certbot/certbot/pull/7707/files#diff-513c9b6861fc981fb2c708db477f542cR7 could also be removed.

Comment (hugovk) [reply pytest.ini]: Thanks, removed!

File: certbot/certbot/_internal/plugins/disco.py [hunk 2]
```diff
@@ -13,6 +13,12 @@
 from certbot import interfaces
 from certbot._internal import constants
 
+try:
+    # Python 3.3+
+    from collections.abc import Mapping  # pylint: disable=no-name-in-module
```
Comment (adferrand) [inline certbot/certbot/_internal/plugins/disco.py]: Disabling `no-name-in-module` is not needed here, since we are running the lint checks on Python 3.

Comment (hugovk) [reply certbot/certbot/_internal/plugins/disco.py]: Thanks, updated!

**Pull request merged**